vm/src/builtins.rs

```diff
@@ -818,7 +822,7 @@ pub fn builtin_build_class_(vm: &mut VirtualMachine, mut args: PyFuncArgs) -> Py
     let mut metaclass = args.get_kwarg("metaclass", vm.get_type());
 
     for base in bases.clone() {
-        if objtype::issubclass(&base.typ(), &metaclass) {
+        if objtype::real_issubclass(vm, &base.typ(), &metaclass)? {
```

Comment:
Also here, pretty sure we shouldn't use the overloaded `__subclasscheck__` and just call `issubclass()` directly.

With this contrived code:
```

class AlwaysGood(type):
	def __subclasscheck__(self, x):
		return True

class M(type): pass
class A(metaclass=M): pass

class M2(type, metaclass=AlwaysGood): pass
class C(A, metaclass=M2): pass
```
RustPython after this patch accepts this code, but CPython seems to have ignored overloaded method and still reports the metaclass conflict - which IMO is the correct behavior.

In general, looks like it'll be tricky to manage which calls should use builtin method, and which accept the overloaded method. Probably the best bet will be to hope that probably-imported-in-the-future CPython test suite will catch the inconsistencies :/

Reply:
You are correct this shouldn't use the real version, I will create a PR soon to fix this.